src/library/module.cpp

```diff
@@ -45,9 +45,11 @@ static char const * g_olean_header   = "oleanfile!!!!!!!";
 
 extern "C" object * lean_save_module_data(b_obj_arg fname, b_obj_arg mod, b_obj_arg mdata, object *) {
     std::string olean_fn(string_cstr(fname));
+    // we first write to a temp file and then move it to the correct path (possibly deleting an older file)
+    // so that we neither expose partially-written files nor modify possibly memory-mapped files
```

Comment:
@leodemoura @tydeu Correct replacement of .olean files is now handled right inside Lean so that build systems do not have to worry about it. This should also obviate the need for file locking.

Reply:
TBD: replacement on Windows, where memory-mapped files cannot be deleted...